scripts/tools/zap_regen_all.py

```diff
@@ -269,8 +284,15 @@ def main():
         if args.run_bootstrap:
             subprocess.check_call(os.path.join(CHIP_ROOT_DIR, "scripts/tools/zap/zap_bootstrap.sh"), shell=True)
 
-        for target in targets:
-            target.generate()
+        if args.parallel:
+            # Ensure each zap run is independent
+            os.environ['ZAP_TEMPSTATE'] = '1'
+            with multiprocessing.Pool() as pool:
+                for _ in pool.imap_unordered(_ParallelGenerateOne, targets):
```

Comment:
IMO simple `lambda x: x.generate()` will suffice here Otherwise, I would use `_ParallelGenerateOne()` (maybe with different name) in the non-multiprocessing case as well for consistency.

Reply:
Cannot seem to do that, I get:

```
AttributeError: Can't pickle local object 'main.<locals>.<lambda>'
```

Reply:
Regarding using it in the non-parallel case: I think it makes the code more complex.

Reply:
> Can't pickle local object 'main.<locals>.<lambda>'

Yayks... OK, so let it be like it is right now.